File: src/repositories/claim-choices.js
// Append-only log (claim_choices) — one row per successful coin claim,
// capturing the EazeScore balance that was converted at claim time. user_id
// is the real Eaze platform user id (see login-logs.js); only recorded when
// the caller actually has one (banner-entered users), never fabricated for
// the typed-phone dev login path.
import { dbEnabled, query } from '../db/client.js';
import { istNowParts } from '../lib/ist-time.js';

export async function recordClaim({ userId, phoneNumber, eazescoreClaimed }) {
  if (!dbEnabled) return { persisted: false };
  const { date, time } = istNowParts();
  const { rows } = await query(
    `INSERT INTO claim_choices (user_id, phone_number, eazescore_claimed, log_date, log_time)
     VALUES ($1, $2, $3, $4, $5)
     RETURNING id, user_id, phone_number, eazescore_claimed, log_date, log_time`,
    [userId, phoneNumber, eazescoreClaimed, date, time],
  );
  return { persisted: true, log: rows[0] };
}

File: app/repositories/claim_choices.py
# Append-only log (claim_choices) — one row per successful coin claim,
# capturing the EazeScore balance that was converted at claim time. user_id
# is the real Eaze platform user id (see login_logs.py); only recorded when
# the caller actually has one (banner-entered users), never fabricated for
# the typed-phone dev login path.
from ..db.client import DB_ENABLED, query
from ..lib.ist_time import ist_now_parts


async def record_claim(*, user_id, phone_number, eazescore_claimed):
    if not DB_ENABLED:
        return {"persisted": False}
    parts = ist_now_parts()
    result = await query(
        """INSERT INTO claim_choices (user_id, phone_number, eazescore_claimed, log_date, log_time)
           VALUES (%s, %s, %s, %s, %s)
           RETURNING id, user_id, phone_number, eazescore_claimed, log_date, log_time""",
        [user_id, phone_number, eazescore_claimed, parts["date"], parts["time"]],
    )
    return {"persisted": True, "log": result["rows"][0]}
